xarray/tests/test_formatting.py

```diff
@@ -134,11 +134,13 @@ def test_format_array_flat(self):
         expected = "0 1 2 ... 98 99"
         assert expected == actual
 
+        # NB: Probably not ideal; an alternative would be cutting after the
+        # first ellipsis
         actual = formatting.format_array_flat(np.arange(100.0), 11)
-        expected = "0.0 ... 99.0"
+        expected = "0.0 ... ..."
```

Comment:
I'd cut after the first ellipsis, two of them in a row look strange to me.

Reply:
Yeah, but what's the rule? Backtrack and see whether the previous values were `...`?

Reply:
something like that, yes:
```python
    if len(pprint_str) > max_width:
        pprint_str = pprint_str[: max(max_width - 3, 0)].rstrip()
        if not pprint_str.endswith("..."):
            pprint_str = pprint_str + "..."
```
Does that seem reasonable?

Reply:
Something like this is very reasonable, for sure. Just a tradeoff of complexity vs. results. 

For example, this case wouldn't change as a result of the function's change, because it ends with a space... We could add that logic in though

Reply:
true, but we could also run `rstrip` first, truncate the string and run `rstrip` again (apologies if that's what you meant, I'm not really sure).

My suggestion might result in something like `783759.270 ... 592000...` but avoids `0.0 ... ...`. If we'd also like to avoid the former, we could not use `"".join()` immediately but instead save the iterable, check if the result of `"".join()` is short enough and remove the last element of the iterable if it isn't.

Reply:
true, but we could also run `rstrip` first, truncate the string and run `rstrip` again (apologies if that's what you meant, I'm not really sure).

My suggestion might result in something like `783759.270 ... 592000...` but avoids `0.0 ... ...`. If we'd also like to avoid the former, we could not use `"".join()` immediately but instead save the iterable, check if the result of `"".join()` is short enough and remove the last element of the iterable if it isn't.

Reply:
Yes 100%, I agree that's better. I'm probably a bit less motivated to make these stat as I feel like the egregious tails have been limited, which was the pressing issue. But can come back to it in the future if no one else takes it up.